### ISSUE TITLE
Problem with in fluid generated arrays TYPO3v8+

### ISSUE THREAD

**Issue by monosize** [FluidTYPO3/vhs#1224]
create an array in a fluid template with ```v:variable.set```

```
<v:variable.set name="conf" value="{0:''}"/>
<f:if condition="{sameheight}">
	<v:variable.set name="conf.1" value="same height"/>
</f:if>
<f:if condition="{mode} > 0">
	<v:variable.set name="conf.2" value="max width"/>
</f:if>
<f:if condition="{samecollapseheight}">
	<v:variable.set name="conf.3" value="collapse"/>
</f:if>
<f:if condition="{everycolumn}">
	<v:variable.set name="conf.4" value="on every column"/>
</f:if>
 ```

the following example works in TYPO3 7.6 but not in TYPO3v8+
```
{conf -> v:iterator.slice(start:1) -> v:iterator.implode(glue:', ')}
```
in TYPO3v8+ an error is displayed ```#1476107295: PHP Warning: htmlspecialchars() expects parameter 1 to be string, array given in```

PHP Code in fluid_template is 
```php
$renderChildrenClosure738 = function() use ($renderingContext, $self) {
$array739 = array (
);return htmlspecialchars($renderingContext->getVariableProvider()->getByPath('conf', $array739), ENT_QUOTES);
};
```
the htmlspecialschars here is wrong 

If I create an array 'conf' in PHP and assign this to the view then there is no error!
```php
$this->view->assign('conf', array('a','b','c','d'));
```

@NamelessCoder or is this a TYPO3/Fluid error

**Comment by NamelessCoder** [comment]
https://github.com/FluidTYPO3/vhs/blob/development/Classes/ViewHelpers/Iterator/SliceViewHelper.php#L26 needs `$escapeOutput = false` - that should solve it. There may be others...

**Comment by monosize** [comment]
ok, I will check this.

**Comment by monosize** [comment]
The result was tested with ```$escapeOutput = false``` in the SliveViewHelper.
Sorry. I forgot to mention that.

It looks like a problem that the array is generated inside a fluid template with ```<v:variable.set .../>```

this is not working:
```
<v:variable.set name="conf" value="{0:'',1:'a',2:'b'}"/>
{conf -> v:iterator.slice(start:1) -> v:iterator.implode(glue:', ')}
```

is the array assigned to the view in the controller -> no error.

**Comment by NamelessCoder** [comment]
What gets assigned as value when you use the above syntax? Something not an array...?

**Comment by monosize** [comment]
It's an array
```
Extbase Variable Dump
array(3 items)
   0 => '' (0 chars)
   1 => 'a' (1 chars)
   2 => 'b' (1 chars)
```

but the syntax {conf -> ... ) is not working
in TYPO3 7.6 there are no problems with this syntax.

**Comment by NamelessCoder** [comment]
Sorry to be away, there's a bit of pressure before feature freeze on TYPO3 8.6.

Try adding `$escapeChildren = false` on those as well. That should output anything passed via render children closure, as a raw variable accessor instead of one wrapped in htmlspecialchars.

Like with `$escapeOutput` this may be missing on other classes as well.

**Comment by monosize** [comment]
```$escapeChildren = false``` is the solution ;-) 
I'll check this on all vhs and flux ViewHelpers.

**Comment by monosize** [comment]
patched for IteratorViewHelpers in this PR #1226

**Comment by bjo3rnf** [comment]
Closing after merge of #1226
